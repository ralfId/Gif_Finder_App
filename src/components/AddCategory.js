import React, { useState } from "react";
import PropTypes from 'prop-types'


export const AddCategory = ({setCategory}) => {
  const [inputValue, setInputValue] = useState('');

  const handleOnChangeInput = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) =>{
      e.preventDefault();
      console.log('handleSubmit', inputValue);
      if ( inputValue.trim().length > 2 ) {
        setCategory(catg => [inputValue, ...catg] );
      }
      setInputValue('');
  };

  return (
    <form onSubmit={handleSubmit}>
        <p>{inputValue}</p>
        <input onChange={handleOnChangeInput} value={ inputValue } type="text" />

    </form>
  );
};

AddCategory.propTypes = {
    setCategory: PropTypes.func.isRequired
}
