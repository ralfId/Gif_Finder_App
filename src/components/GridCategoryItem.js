import React from 'react'
import PropTypes from 'prop-types';

export const GridCategoryItem = ({ title, img }) => {
    return (
        <div className='card animate__fadeInLeft'>
            <img src={img} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GridCategoryItem.protoTypes = {
    title: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired
};