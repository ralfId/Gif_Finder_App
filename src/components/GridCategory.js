import React, { useState, useEffect } from 'react'
import { GetGif } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GridCategoryItem } from './GridCategoryItem';

export const GridCategory = ({ category }) => {


    // const [gifImg, setGifImg] = useState([])
    const {data:gifImg, loading} = useFetchGifs(category);
    
    // useEffect(() => {
    //     GetGif(category).then(setGifImg);
    // }, [category]);

    

    return (
        <>
            <h3>{category}</h3>
            {loading && <p>Loading...</p>}
            <div className='card-container animate__zoomIn'>
                {
                    gifImg.map(img => (
                        <GridCategoryItem
                            key={img.id} {...img} />
                    ))
                }
            </div>


        </>
    )
}
