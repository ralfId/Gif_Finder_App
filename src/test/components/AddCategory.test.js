import React from 'react'
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";
import { Simulate } from 'react-dom/test-utils';

describe('AddCategory.js tests', () => {
    
    const setCategory = jest.fn();

    let  wrapper;


    //se ejecuta todo lo que se necesita reinicializar ante cada test
    beforeEach(()=>{
        jest.clearAllMocks(); //se limpian todas las simulaciones
        wrapper  = shallow(<AddCategory setCategory= {setCategory} />);
    });

    test('Pueva en <AddCategory />', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de cambiar el texto del <input/> en <AddCategory />', () => {
        
            const input = wrapper.find('input');
            const value = 'Hola mundo';

            input.simulate('change', {target:{value}});
            
            expect(wrapper.find('p').text().trim()).toBe(value);
    });
    
    test('No debe realizar nada al realizar el submit en el <form />', () => {
        //simulando un submit
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        
        expect(setCategory).not.toHaveBeenCalled();
    });
    
    test('Debe de llamar el SetCategory y limpiar el <input/> al simular el onSubmit', () => {
        
        
        const value = "categoria de busqueda";

        //simular el inputChange
        wrapper.find('input').simulate('change', {target:{value}});
        expect(wrapper.find('p').text().trim()).toBe(value);

        //simular el submit
        wrapper.find('form').simulate('submit', {preventDefault(){}})
        //setCategory debe llamarse almenos una vez
        expect(setCategory).toHaveBeenCalledTimes(1);
        expect(setCategory).toHaveBeenLastCalledWith(expect.any(Function))//verifica que se ejecuta, pero se espere sea una funcion 
        //el valor del <input/> debe estas "" (vacio)
        expect(wrapper.find('input').prop('value')).toBe('');
    })
    
    
});