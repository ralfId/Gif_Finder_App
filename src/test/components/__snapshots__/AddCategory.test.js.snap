// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory.js tests Pueva en <AddCategory /> 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
